chain/network/src/peer_manager/peer_manager_actor.rs

```diff
@@ -852,7 +853,26 @@ impl PeerManagerActor {
                     None
                 }
             })
-            .collect::<Vec<_>>()
+            .collect::<Vec<_>>();
+
+        // remove peers to which we have no route from the list
```

Comment:
what is the scenario when it happens? 

can we add a test for this case?

Reply:
I don't know exactly why this happens yet. This will require some time to investigate the logic. I think it's a good idea to add a print to alert if that happens again, while we work on finding the issue.

For example:
- there may be an issue with routing table not being updated properly. Or for whatever other reason why `active_peers` has peers to which we don't have a route to. 

Anyway, if the issue is in this function, that would explain the error we are seeing,